h5netcdf/core.py

```diff
@@ -125,28 +132,55 @@ def shuffle(self):
 
 NOT_A_VARIABLE = b'This is a netCDF dimension but not a netCDF variable.'
 
+class _LazyObjectLookup(Mapping):
+    def __init__(self, parent, object_cls):
+        self._parent = parent
+        self._object_cls = object_cls
+        self._objects = OrderedDict()
+
+    def __setitem__(self, name, object):
+        self._objects[name] = object
+
+    def add(self, name):
+        self._objects[name] = None
+
+    def __iter__(self):
+        for name in self._objects:
+            yield name
+
+    def __len__(self):
+        return len(self._objects)
+
+    def __getitem__(self, key):
+        if self._objects[key] != None:
```

Comment:
It's a best practice to use `is not None` instead of `!= None`: https://www.python.org/dev/peps/pep-0008/#programming-recommendations